### DISCUSSION TITLE
How to aggregate joined column?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#694] [Q&A] [answered]
Given the below table with just the `ProductId` and an array `HistoricalPrices`:

![image](https://user-images.githubusercontent.com/35210495/191401799-e30a1ff4-44da-4a13-a40a-e6bb596f1109.png)

What is being summed when we do a `measures["historicalPrices.SUM"] = tt.agg.sum(historicalPrices["HistoricalPrice"])`?

**Comment by HuifangYeo** [comment]
This is the schema:
![image](https://user-images.githubusercontent.com/35210495/191402283-7cdfb178-dc2c-4808-83ae-ba7a6c9d2ea4.png)

**Comment by Skyscimitar** [comment, marked as answer]
Hi HuiFang,
Since you are aggregating a table column, it is aggregated on all levels by default. I'm assuming that there are mulitple `OrderId` per `ProductId` so the value you are getting for `historicalPrices.SUM` is probably the sum of the historical price for a product over all the `OrderId` it appears in.
To avoid this, you should use:
```python
tt.agg.sum(tt.agg.single_value(table["historicalPrices"]), scope=tt.OriginScope(l["ProductId"]))
```

**Reply by jackschd** [reply]
Thankyou for your answer. This is indeed the intention, and the correct implementation.
